### ISSUE TITLE
[Camera Support]: green fragments on Clips of only one Cam - but only on Android

### ISSUE THREAD

**Issue by Bodengriller** [blakeblackshear/frigate#7456]
### Describe the problem you are having

I have three cameras. (2x yi-home 9FUS/y20ga 1x yi-home IFUS/y211ga)
Everything works well so far.
However, on one camera ("Parkplatz" - the IFUS/y211ga), the recorded clips cannot be played back from an Android-smartphone (Samsung S21).
I only see green fragmentations there.
![image](https://github.com/blakeblackshear/frigate/assets/50927649/58dff2f3-3d74-4e1b-9cee-cb4291df62b6)
(Tested with the following browsers: Chrome, Firefox, Brave).
There is no problem with live-stream or Birdseye.
If I open the clips via the web browser on the computer, everything works fine. (Chrome)
With the other two cameras, the recordings can also be played from the smartphone.
I suspect there is some format error here that Chrome on the computer can compensate for, but the browsers on the smartphone cannot.
Any idea what setting I could test for this camera to make the clips playable from the smartphone as well?

### Version

0.12.1-367d724

### Frigate config file

```yaml
detectors:
  coral:
    type: edgetpu
    device: pci

#activate hardware acceleration
ffmpeg:
  hwaccel_args: preset-vaapi
  #hwaccel_args: preset-intel-qsv-h264

mqtt:
  enabled: True
  host: 192.168.178.152
  port: 1883
  user: "xxx"
  password: "xxx"

rtmp:
  enabled: False
  
##########----------General ui----------##########
birdseye:
  # Optional: Enable birdseye view (default: True)
  enabled: True
  # Optional: Restream birdseye via RTSP (default: False)
  # NOTE: Enabling this will set birdseye to run 24/7 which may increase CPU usage somewhat.
  restream: False
  # Optional: Width of the output resolution (default: 1280)
  width: 640
  # Optional: Height of the output resolution (default: 720)
  height: 360
  # Optional: Encoding quality of the mpeg1 feed (default: 8)
  # 1 is the highest quality, and 31 is the lowest. Lower quality feeds utilize less CPU resources.
  quality: 8
  # Optional: Mode of the view. Available options are: objects, motion, and continuous
  #   objects - cameras are included if they have had a tracked object within the last 30 seconds
  #   motion - cameras are included if motion was detected in the last 30 seconds
  #   continuous - all cameras are included always
  mode: continuous

live:
  height: 360

timestamp_style:
    position: "br"
    format: "%d.%m.%Y %H:%M:%S"
    effect: solid

##########----------General-Config----------##########
objects:
  track:
    - person
    - car
    - bicycle
    - motorcycle
  filters:
    person:
      # Optional: minimum score for the object to initiate tracking (default: 0.5)
      min_score: 0.4
      # Optional: minimum decimal percentage for tracked object's computed score to be considered a true positive (default: 0.7)
      threshold: 0.7
    car:
      # Optional: minimum score for the object to initiate tracking (default: 0.5)
      min_score: 0.5
      # Optional: minimum decimal percentage for tracked object's computed score to be considered a true positive (default: 0.7)
      threshold: 0.7

snapshots:
  #enabled: False
  enabled: True
  clean_copy: False
  timestamp: True
  bounding_box: True
  retain:
    default: 10
    objects:
      person: 30

# only keeps Motion-Clips for 10 days (default)
record:     
  #enabled: False
  enabled: True
  retain:
    days: 7
    mode: all
  events:
    pre_capture: 15
    post_capture: 30
    retain:
      default: 14
      mode: active_objects

##########----------Camera-Config----------##########
cameras:
  #####------Kamera Carport-----#####
  Carport:
    #enabled: False
    ui:
      order: 1
    ffmpeg:
      input_args: -rtsp_transport tcp
      inputs:
        - path: rtsp://192.168.178.41/ch0_1.h264
          roles:
            - detect
        - path: rtsp://192.168.178.41/ch0_1.h264
          roles:
            - record
    detect:
      #enabled: False
      enabled: True
      width: 640
      height: 360
      fps: 5
    motion:
      mask:
      - 0,360,0,217,382,360
      - 0,360,0,0,105,0,153,360
      - 0,0,203,0,310,66,0,180
      - 640,0,197,0,304,63,307,78,640,64
      - 640,0,640,360,499,360,521,190,517,0
      - 640,360,640,264,417,257,305,299,205,360
  #####-------------------------#####

  #####-----Kamera Parkplatz----#####
  Parkplatz:
    #enabled: False
    ui:
      order: 2
    ffmpeg:
      input_args: -rtsp_transport tcp
      inputs:
        - path: rtsp://192.168.178.40/ch0_1.h264
          roles:
            - detect
        - path: rtsp://192.168.178.40/ch0_1.h264
          roles:
            - record
    detect:
      #enabled: False
      enabled: True
      width: 640
      height: 360
      fps: 5
    motion:
      mask:
      - 640,187,377,360,640,360
      - 640,0,640,360,604,360,427,0
      - 640,175,511,154,345,124,349,96,216,78,0,104,0,33,0,0,640,0
      - 588,0,0,0,0,124
      - 101,360,0,360,0,0,80,0
      - 0,360,346,360,267,291,0,291
  #####-------------------------#####

  #####----Kamera Stellplatz----#####
  Stellplatz:
    #enabled: False
    ui:
      order: 3
    ffmpeg:
      input_args: -rtsp_transport tcp
      inputs:
        - path: rtsp://192.168.178.42/ch0_1.h264
          roles:
            - detect
        - path: rtsp://192.168.178.42/ch0_1.h264
          roles:
            - record
    detect:
      #enabled: False
      enabled: True
      width: 640
      height: 360
      fps: 5
    motion:
      mask:
      - 0,360,0,0,113,0,177,360
      - 640,360,640,65,361,360
      - 492,0,640,0,640,360,442,360
      - 0,0,0,44,318,47,640,80,640,0
      - 0,0,400,0,314,49,0,136
  #####-------------------------#####
```

### Relevant log output

```shell
2023-08-06 20:27:26.365443470  [INFO] Starting Frigate...
2023-08-06 20:27:30.624650772  [2023-08-06 20:27:30] frigate.app                    INFO    : Starting Frigate (0.12.1-367d724)
2023-08-06 20:27:30.712156130  [2023-08-06 20:27:30] peewee_migrate                 INFO    : Starting migrations
2023-08-06 20:27:30.726506529  [2023-08-06 20:27:30] peewee_migrate                 INFO    : There is nothing to migrate
2023-08-06 20:27:30.767701005  [2023-08-06 20:27:30] frigate.app                    INFO    : Output process started: 292
2023-08-06 20:27:30.786041940  [2023-08-06 20:27:30] frigate.app                    INFO    : Camera processor started for Carport: 296
2023-08-06 20:27:30.800413837  [2023-08-06 20:27:30] detector.coral                 INFO    : Starting detection process: 291
2023-08-06 20:27:30.800422187  [2023-08-06 20:27:30] frigate.detectors.plugins.edgetpu_tfl INFO    : Attempting to load TPU as pci
2023-08-06 20:27:30.807077497  [2023-08-06 20:27:30] frigate.app                    INFO    : Camera processor started for Parkplatz: 298
2023-08-06 20:27:30.854289545  [2023-08-06 20:27:30] frigate.detectors.plugins.edgetpu_tfl INFO    : TPU found
2023-08-06 20:27:30.856946893  [2023-08-06 20:27:30] frigate.app                    INFO    : Camera processor started for Stellplatz: 300
2023-08-06 20:27:30.856955583  [2023-08-06 20:27:30] frigate.app                    INFO    : Capture process started for Carport: 302
2023-08-06 20:27:30.858293326  [2023-08-06 20:27:30] frigate.app                    INFO    : Capture process started for Parkplatz: 315
2023-08-06 20:27:30.882389459  [2023-08-06 20:27:30] frigate.app                    INFO    : Capture process started for Stellplatz: 318
```

### FFprobe output from your camera

```shell
[{"return_code":0,"stderr":"","stdout":{"programs":[],"streams":[{"avg_frame_rate":"20/1","codec_long_name":"H.264/AVC/MPEG-4AVC/MPEG-4part10","height":360,"width":640}]}},{"return_code":0,"stderr":"","stdout":{"programs":[],"streams":[{"avg_frame_rate":"20/1","codec_long_name":"H.264/AVC/MPEG-4AVC/MPEG-4part10","height":360,"width":640}]}}]

(same output on all three cams)
```

### Operating system

Proxmox

### Install method

Docker Compose

### Coral version

M.2

### Network connection

Wired

### Camera make and model

yi-home IFUS/y211ga (yi-hack 0.2.7)

**Comment by NickM-27** [comment]
What settings are available for the camera?

**Comment by Bodengriller** [comment]
Not very much:
![image](https://github.com/blakeblackshear/frigate/assets/50927649/f8a6f384-b69b-4bcf-83d1-6c934daa55fe)
(same settings as the two other cams, which are working fine)

**Comment by NickM-27** [comment]
In my experience issues like this are par for the course when using cheaper & hacked cameras. Given that it seems there is not even control of the pixel format, I don't see anything that would make a difference here.

**Comment by Bodengriller** [comment]
Hehe ... yes, that the cameras are cheap junk I am aware :D
I'm just wondering that it works from the browser from the computer, but not from the smartphone.
I had hope that there might still be some ffmpeg settings that could help.

Also curious is that the live stream works.

Are there any log files that could be used to find out what the problem is in the first place?
Could I somehow download one of the working clips from the other two cameras and analyze it in some way so that we could compare what the error is here in the first place?
Maybe it's not a big thing.

**Comment by Bodengriller** [comment]
Just noticed by chance:
The clips can also be played correctly with a Pocophone X2.
The same problem with the green fragments occurs with a Samsung A52 as with the S21.
Interesting. 
So Samsung does not like something about the clips from this one camera.

**Comment by NickM-27** [comment]
I don't expect that any ffmpeg args short of re-encoding the stream will change this

**Comment by Bodengriller** [comment]
Thanks for the feedback.
I do not check it.
I have tried a few things in the meantime. First I downloaded two clips (one from a camera that works and one from the camera whose clips display strangely on Samsung smartphones).
I have tested various players - also VLC for Android...which can play more or less any crap.
The phenomenon is always the same.
If I open the seemingly broken clip on the PC in VLC (where the clip is played correctly) and simply re-save it once, then play it again on the samsung smartphone, it is also played correctly there.
Very curious... (especially since these problems do not exist on the Pocophone).
I don't know what else I could test or do - I'll probably have to deal with it.
